eo-maven-plugin/src/main/java/org/eolang/maven/ProbeMojo.java

```diff
@@ -0,0 +1,246 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import com.jcabi.log.Logger;
+import com.jcabi.xml.XML;
+import com.jcabi.xml.XMLDocument;
+import com.yegor256.tojos.Tojo;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.TreeSet;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.cactoos.iterable.Filtered;
+import org.cactoos.list.ListOf;
+import org.eolang.maven.hash.ChNarrow;
+import org.eolang.maven.hash.ChResolve;
+import org.eolang.maven.hash.CommitHash;
+import org.eolang.maven.objectionary.Objectionary;
+import org.eolang.maven.objectionary.OyFallbackSwap;
+import org.eolang.maven.objectionary.OyHome;
+import org.eolang.maven.objectionary.OyRemote;
+import org.eolang.maven.util.Online;
+import org.eolang.maven.util.Rel;
+
+/**
+ * Go through all `probe` metas in XMIR files, try to locate the
+ * objects pointed by `probe` in Objectionary and if found register them in
+ * catalog.
+ *
+ * @since 0.28.11
+ * @checkstyle CyclomaticComplexityCheck (300 lines)
+ */
+@Mojo(
+    name = "probe",
+    defaultPhase = LifecyclePhase.PROCESS_SOURCES,
+    threadSafe = true
+)
+public final class ProbeMojo extends SafeMojo {
+
+    /**
+     * The Git hash to pull objects from, in objectionary.
+     *
+     * @since 0.21.0
+     */
+    @SuppressWarnings("PMD.ImmutableField")
+    @Parameter(property = "eo.tag", required = true, defaultValue = "master")
+    private String tag = "master";
+
+    /**
+     * Target directory.
+     *
+     * @checkstyle MemberNameCheck (7 lines)
+     */
+    @Parameter(property = "eo.home")
+    @SuppressWarnings("PMD.ImmutableField")
+    private Path outputPath = Paths.get(System.getProperty("user.home")).resolve(".eo");
+
+    /**
+     * Read hashes from local file.
+     *
+     * @checkstyle MemberNameCheck (7 lines)
+     */
+    @Parameter(property = "offlineHashFile")
+    private Path offlineHashFile;
+
+    /**
+     * Return hash by pattern.
+     * -DofflineHash=0.*.*:abc2sd3
+     * -DofflineHash=0.2.7:abc2sd3,0.2.8:s4se2fe
+     *
+     * @checkstyle MemberNameCheck (7 lines)
+     */
+    @Parameter(property = "offlineHash")
+    private String offlineHash;
+
+    /**
+     * The objectionary.
+     */
+    @SuppressWarnings("PMD.ImmutableField")
+    private Objectionary objectionary;
+
+    @Override
+    public void exec() throws IOException {
+        if (!new Online().value()) {
+            Logger.warn(
+                this, "There is not internet connection. Probe skipped"
```

Comment:
@MikhailLipanin Better "We are offline: skipping probes".

Reply:
done